platforms/fabric/src/main/java/com/dfsek/terra/fabric/generation/FabricChunkGeneratorWrapper.java

```diff
@@ -82,19 +86,23 @@ public FabricChunkGeneratorWrapper(TerraBiomeSource biomeSource, long seed, Conf
         this.seed = seed;
     }
     
+    public Registry<StructureSet> getNoiseRegistry() {
+        return noiseRegistry;
+    }
+    
     @Override
     protected Codec<? extends net.minecraft.world.gen.chunk.ChunkGenerator> getCodec() {
-        return Codecs.CODEC;
+        return Codecs.FABRIC_CHUNK_GENERATOR_WRAPPER;
     }
     
     @Override
     public net.minecraft.world.gen.chunk.ChunkGenerator withSeed(long seed) {
-        return new FabricChunkGeneratorWrapper((TerraBiomeSource) this.biomeSource.withSeed(seed), seed, pack, settingsSupplier);
+        return new FabricChunkGeneratorWrapper(noiseRegistry, (TerraBiomeSource) this.biomeSource.withSeed(seed), seed, pack, settings);
     }
     
     @Override
     public MultiNoiseUtil.MultiNoiseSampler getMultiNoiseSampler() {
-        return (x, y, z) -> new MultiNoiseUtil.NoiseValuePoint(0, 0, 0, 0, 0, 0);
+        return MultiNoiseUtil.method_40443(); // zero
```

Comment:
Add comment indicating what this method is

Reply:
there is already a comment.

Reply:
Ah, it wasn't clear